testing/citest/spinnaker_testing/dcos_contract.py

```diff
@@ -0,0 +1,155 @@
+      return []
+
+    decoder = json.JSONDecoder()
+    try:
+      doc = decoder.decode(dcos_response.output)
+      if not isinstance(doc, list):
+        doc = [doc]
+      self.filter_all_objects_to_observation(context, doc, observation)
+    except ValueError as vex:
+      error = 'Invalid JSON in response: %s' % str(dcos_response)
+      logging.getLogger(__name__).info('%s\n%s\n----------------\n',
+                                       error, traceback.format_exc())
+      observation.add_error(jp.JsonError(error, vex))
+      return []
+
+    return observation.objects
+
+
+class DcosObjectFactory(object):
+  # pylint: disable=too-few-public-methods
+
+  def __init__(self, dcoscli):
+    self.__dcoscli = dcoscli
+
+  def new_get_marathon_resources(self, type, action, extra_args=None):
+    """Specify a resource list to be returned later.
+
+    Args:
+      type: dcos's name for the DC/OS resource type.
+
+    Returns:
+      A jc.ObjectObserver to return the specified resource list when called.
+    """
+    if extra_args is None:
+      extra_args = []
+
+    cmd = self.__dcoscli.build_dcoscli_command_args(
+        subcommand='marathon', resource=type, action=action, args=['--json'] + extra_args)
+    return DcosObjectObserver(self.__dcoscli, cmd)
+
+
+class DcosClauseBuilder(jc.ContractClauseBuilder):
+  """A ContractClause that facilitates observing DC/OS state."""
+
+  def __init__(self, title, dcoscli, retryable_for_secs=0, strict=False):
+    """Construct new clause.
+
+    Args:
+      title: The string title for the clause is only for reporting purposes.
+      dcoscli: The DcosCliAgent to make the observation for the clause to
+         verify.
+      retryable_for_secs: Number of seconds that observations can be retried
+         if their verification initially fails.
+      strict: DEPRECATED flag indicating whether the clauses (added later)
+         must be true for all objects (strict) or at least one (not strict).
+         See ValueObservationVerifierBuilder for more information.
+         This is deprecated because in the future this should be on a per
+         constraint basis.
+    """
+    super(DcosClauseBuilder, self).__init__(
+        title=title, retryable_for_secs=retryable_for_secs)
+    self.__factory = DcosObjectFactory(dcoscli)
+    self.__strict = strict
+
+  def get_marathon_resources(self, type, extra_args=None, no_resource_ok=False):
+    """Observe resources of a particular type.
+
+    This ultimately calls a "dcos marathon |type| |extra_args|"
+
+    Args:
+      no_resource_ok: Whether or not the resource is required.
+          If the resource is not required, "not found" is treated as a valid
+          check. Because resource deletion is asynchronous, there is no
+          explicit API here to confirm that a resource does not exist.
+    """
+    self.observer = self.__factory.new_get_marathon_resources(
+        type, action='list', extra_args=extra_args)
+
+    if no_resource_ok:
```

Comment:
There weren't any cases where we were actually setting `no_resource_ok` so I removed the parameter and left the rest of the test as is.